### ISSUE TITLE
Pre-configuring Multiple Responses for a Conversation

### ISSUE THREAD

**Issue by dvirginz** [blip-solutions/promptwatch-client#17]
Hello `Proptwatch` maintainers and community,

I am currently working on a project where I am building a chatbot to assist users in buying a product. I want to be able to evaluate the bot's responses in various conversation flows, and I was wondering if the `Proptwatch` library supports such a use case.

Here are the three specific flows I'd like to test:

1. **Positive Flow:** The user agrees with everything and always responds positively, essentially saying 'yes' to all prompts.
2. **Inquisitive Flow:** The user asks 1 to 3 out of 5 possible questions during the interaction.
3. **Human Representative Request:** Throughout the conversation, the user consistently asks to speak to a human representative.

Ideally, I would like to set up a mock "user" (which could be another bot) to communicate with the bot we're aiming to send to production. This would simulate these three scenarios and allow us to test our bot's responses.

**Questions:**
1. Does the `Proptwatch` library support this use case of pre-configuring multiple conversation flows?
2. If yes, could you provide any pointers or documentation links on how to set it up?
3. If no, do you have any plans in the future roadmap to support such functionality or do you know of any other tools/libraries that might assist with this?

Thank you for your time and looking forward to your response!

**Comment by ju-bezdek** [comment]
Hey... 
it is not publicly available for now but we've been working on something like this...

Here is how it works...:

1. It asks you what kind of chatbot are you building and some context data
2. It will generate test personas .... you can change them, give feedback how to regenerate them
3. for each personas it generates a topic / problem / goal that this type of persona might have ... again, something you can modify

Based on this, we are generating virtual AI-powered users, that simulate the users messages

here are some screenshots...

<img width="904" alt="image" src="https://github.com/blip-solutions/promptwatch-client/assets/27913091/833935eb-a849-4df8-81c3-a6042b716229">

Here is the simulated conversation


the "User messages" here are generated by PromptWatch
<img width="1505" alt="image" src="https://github.com/blip-solutions/promptwatch-client/assets/27913091/251e69f9-c011-4c5c-bb84-c80ccd8c4171">


Is this something like you had in mind?

**Comment by dvirginz** [comment]
That's exactly what I'm looking for.

Optimally, the other unit tests will be evaluated based on these answers.

The incentive is to have a true CI/CD pipeline for LLMs. This way, we can push PRs and know that at least the flows we define as critical are still working.

**Comment by ju-bezdek** [comment]
> Optimally, the other unit tests will be evaluated based on these answers.

Yes, that is the idea, although currently it's expected that you'd validate these autogenerated scenarios, to populate the verified test cases.

As I mentioned, this is currently not publicly available. Are you interested in early access? Feel free to reach out via email: juraj.bezdek@gmail.com and we can discuss this in more detail.
